server/src/main/java/org/elasticsearch/common/geo/builders/GeometryCollectionBuilder.java

```diff
@@ -197,9 +197,6 @@ public Object buildLucene() {
             }
         }
 
-        if (shapes.size() == 1) {
-            return shapes.get(0);
-        }
```

Comment:
why did you remove this?

Reply:
Because for Lucene geometry the shape could be a point defined as a `double[]` which fails downstream casting.